File: DSA/Nivedan_590014904/Day_1_Question_2.c
#include <stdio.h>
void find(int arr[], int n) {
    int duplicate = -1;
    int missing = -1;
    for (int i = 1; i <= n; i++) {
        int count = 0;
        for (int j = 0; j < n; j++) {
            if (arr[j] == i) {
                count++;
            }
        }
        if (count == 2) {
            duplicate = i;
        } else if (count == 0) {
            missing = i;
        }
    }
    printf("Missing: %d, Twice: %d\n", missing, duplicate);
}
int main() {
    int arr[100], n;
    printf("Enter number of elements in array: ");
    scanf("%d", &n);

    printf("Enter %d elements (range: 1 to %d):\n", n, n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    find(arr, n);
    return 0;
}

File: DSA/SuryanshShah_590013441/Day1Question2.java
import java.util.*;
public class Day1Question2 {
    public static void main(String[] args) {
        Scanner scr = new Scanner(System.in);
        System.out.println("Enter the number of elements in the array:");
        int n = scr.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter the elements into the array:");
        for (int i = 0; i < n; i++) {
            arr[i] = scr.nextInt();
        }
        Arrays.sort(arr);
        int duplicate = -1, missing = -1;
        for (int i = 0; i < n - 1; i++) {
            if (arr[i] == arr[i + 1]) {
                duplicate = arr[i];
            } else if (arr[i + 1] - arr[i] > 1) {
                missing = arr[i] + 1;
            }
        }
        if (arr[0] != 1) {
            missing = 1;
        } else if (arr[n - 1] != n) {
            missing = n;
        }
        System.out.println("Duplicate number: " + duplicate);
        System.out.println("Missing number: " + missing);
    }
}
